src/packet/public_key_encrypted_session_key.js

```diff
@@ -83,11 +111,27 @@ class PublicKeyEncryptedSessionKeyPacket {
    */
   write() {
     const arr = [
-      new Uint8Array([this.version]),
-      this.publicKeyID.write(),
+      new Uint8Array([this.version])
+    ];
+
+    if (this.version === 6) {
+      if (this.publicKeyFingerprint !== null) {
```

Comment:
I think it might be easier to follow what's going on by referencing the wildcard keyID?
```suggestion
      if (!this.publicKeyID.isWildcard()) {
```